// dc-react/components/LineChart.jsx
import React, { Component } from 'react';
import dc from 'dc';
import { Base } from './Base';
import PropTypes from 'prop-types';


class LineChart extends Component {
  //updated by WhoIN Lee
  static propTypes = {
    renderArea: PropTypes.bool,
    xAxis: PropTypes.func,
    yAxis: PropTypes.func,
    x: PropTypes.func,
    round: PropTypes.func,
    xUnits: PropTypes.func,
    elasticY: PropTypes.bool,
    renderHorizontalGridLines: PropTypes.bool,
    renderVerticalGridLines: PropTypes.bool,
    brushOn: PropTypes.bool,
    valueAccessor: PropTypes.func,
    stack: PropTypes.func,
  };

  loadChart = (container) => {
    const chart = dc.lineChart(container);
    const helper = this.props.chartHelper(this, chart);
    helper.setProperties('renderArea', 'x', 'round', 'xUnits',
                         'elasticY', 'renderHorizontalGridLines', 'renderVerticalGridLines', 'brushOn',
                         'valueAccessor')
          .setContextProperties('stack');

    if (this.props.xAxis) {
      this.props.xAxis(chart.xAxis());
    }
    if (this.props.yAxis) {
      this.props.yAxis(chart.yAxis());
    }

    chart.render();
  };

  render() {
    return <div className={this.props.className} ref={ this.loadChart }/>;
  }
}

export default Base(LineChart);
